<a href="https://colab.research.google.com/github/rafidfajar/data-science-2026/blob/main/Pertemuan2_Muhamad_Rafid_Fajar_250401020195.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

import library

In [1]:
import numpy as np
import pandas as pd

list

In [2]:
# Membuat list
data = [10, 20, 'Python', True, 3.14]
# Akses & slicing
print(data[0]) # → 10
print(data[-1]) # → 3.14 (dari belakang)
print(data[1:3]) # → [20, 'Python']
# Manipulasi
data.append(99) # tambah di akhir
data.insert(1, 55) # sisipkan di posisi 1
data.remove(20) # hapus nilai pertama = 20

data2 = [10, 20, 30, 40]
data2.sort() # urutkan (jika tipe sama)

10
3.14
[20, 'Python']


Dictonary

In [3]:
mahasiswa = {
'nama': 'Muhamad Rafid fajar',
'nim': '250401020195',
'prodi': 'Informatika',
'ipk': 3.85
}
# Akses nilai
print(mahasiswa['nama']) # → 'Budi Santoso'
print(mahasiswa.get('usia', 23)) # default jika key tak ada
# Update dan hapus
mahasiswa['angkatan'] = 2024 # tambah key baru
del mahasiswa['nim'] # hapus key
print(mahasiswa['angkatan'])

Muhamad Rafid fajar
23
2024


NumPy Array

In [4]:
import numpy as np
# Dari list Python
a1D = np.array([1, 2, 3, 4, 5]) # 1D
a2D = np.array([[1,2,3],[4,5,6]]) # 2D
# Array dengan nilai inisial
zeros = np.zeros((3, 4)) # nol semua
ones = np.ones((2, 3)) # satu semua
eye = np.eye(4) # matriks identitas
full = np.full((2, 2), 7) # nilai konstan
# Sequence
arange = np.arange(0, 20, 2) # 0,2,4,...18
linsp = np.linspace(0, 1, 11) # 11 titik merata
# Array acak (reproducible)
rng = np.random.default_rng(seed=42)
rand_int = rng.integers(0, 100, (3, 4))
rand_flt = rng.normal(loc=0, scale=1, size=(100,))

Operasi Vektorisasi dan Fungsi Statistik

In [ ]:
a = np.array([1, 2, 3, 4, 5])
b = np.array([10, 20, 30, 40, 50])
print(a + b) # → [11 22 33 44 55]
print(a * b) # → [10 40 90 160 250]
print(a ** 2) # → [1 4 9 16 25]
# Universal functions (ufuncs)
np.sqrt(a) # akar kuadrat tiap elemen
np.exp(a) # eksponensial (e^x)
np.log(b) # logaritma natural
# Fungsi statistik
data = rng.normal(0, 1, 1000)
print(f'Mean: {np.mean(data):.4f}')
print(f'Std: {np.std(data):.4f}')
print(f'Min: {np.min(data):.4f}')
print(f'Max: {np.max(data):.4f}')
print(f'Q1: {np.percentile(data,25):.4f}')
print(f'Q3: {np.percentile(data,75):.4f}')

Pandas

In [5]:
import pandas as pd
# Series dari list — index default 0,1,2...
s1 = pd.Series([85, 92, 78, 95, 88])
# Series dengan index kustom
nilai = pd.Series(
[85, 92, 78, 95],
index=['Alice','Bob','Caca','Dina']
)
print(nilai['Bob']) # → 92
print(nilai[nilai>80]) # filter
print(nilai.mean()) # → 87.5
print(nilai.describe()) # statistik lengkap

92
Alice    85
Bob      92
Dina     95
dtype: int64
87.5
count     4.000000
mean     87.500000
std       7.593857
min      78.000000
25%      83.250000
50%      88.500000
75%      92.750000
max      95.000000
dtype: float64


Pandas DataFrame

In [6]:
data = {
'nama': ['Ali','Budi','Caca','Dina'],
'usia': [22, 25, 21, 23],
'kota': ['Jakarta','Bandung','Surabaya','Medan'],
'ipk': [3.8, 3.5, 3.9, 3.7]
}
df = pd.DataFrame(data)
# Informasi dasar
print(df.shape) # → (4, 4)
print(df.dtypes) # tipe data per kolom
print(df.columns) # nama-nama kolom

(4, 4)
nama     object
usia      int64
kota     object
ipk     float64
dtype: object
Index(['nama', 'usia', 'kota', 'ipk'], dtype='object')


## **Latihan Pertemuan 2**

In [26]:
import pandas as pd
import numpy as np
# 1 Muat Dataset Titanic
url = 'https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv'

df = pd.read_csv(url)

df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


**Q1. Berapa jumlah baris dan kolom dataset Titanic?**

In [36]:
print("Jumlah baris :", df.shape[0])
print("Jumlah Kolom :", df.shape[1])

print("Nama Kolom :")

for k in df.columns.tolist():
  print("- ",k)

Jumlah baris : 891
Jumlah Kolom : 12
Nama Kolom :
-  PassengerId
-  Survived
-  Pclass
-  Name
-  Sex
-  Age
-  SibSp
-  Parch
-  Ticket
-  Fare
-  Cabin
-  Embarked


**Q2. Kolom mana yang memiliki nilai kosong
(missing)?**

In [62]:
missing = df.isnull().sum()
missing_data = pd.DataFrame({
    'Jumlah': df.isnull().sum(),
     'Persentase (%)': (df.isnull().sum() / len(df)) * 100
})
# Menampilkan hanya yang memiliki missing value
print(missing_data[missing_data['Jumlah'] > 0])

          Jumlah  Persentase (%)
Age          177       19.865320
Cabin        687       77.104377
Embarked       2        0.224467


**Q3. Berapa persen penumpang yang selamat (Survived=1)??**

In [53]:

survival_rate = df['Survived'].mean() * 100
print(f"Persentase Penumpang Selamat: {survival_rate:}%")

Persentase Penumpang Selamat: 38.38383838383838%


**Q4.Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-
rata usianya?**

In [64]:
wk1 = df[(df['Sex']=='female') &
(df['Pclass']==1)]

print("Jumlah penumpang wanita :",len(wk1))
print("Jumlah penumpang wanita Rata-rata usia :",wk1['Age'].mean())

Jumlah penumpang wanita : 94
Jumlah penumpang wanita Rata-rata usia : 34.61176470588235


**Q5. Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas
penumpang. Kelas mana yang paling tinggi?**

In [66]:
cs = df.groupby('Pclass')['Survived'].mean() * 100
print(cs)
print(f"Kelas dengan tingkat keselamatan tertinggi: Kelas {cs.idxmax()}")

Pclass
1    62.962963
2    47.282609
3    24.236253
Name: Survived, dtype: float64
Kelas dengan tingkat keselamatan tertinggi: Kelas 1


**Q6. Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik.
Sertakan interpretasi singkat.**

In [68]:
gender_survival = df.groupby('Sex')['Survived'].mean() * 100
print(gender_survival)

Sex
female    74.203822
male      18.890815
Name: Survived, dtype: float64


terlihat perbedaan drastis di mana tingkat keselamatan wanita jauh lebih tinggi (74%)
dibandingkan pria yang hanya (18%)